watertap/costing/unit_models/clarifier.py

```diff
@@ -0,0 +1,361 @@
+#################################################################################
+"""
+[1] Sharma, Jwala R., Mohammad Najafi, and Syed R. Qasim.
+"Preliminary cost estimation models for construction, operation, and maintenance of water treatment plants."
+Journal of Infrastructure Systems 19.4 (2013): 451-464.
+
+[2] Byun, Jaewon, Maravelias, Christos.
+Benchmark Model for Wastewater Treatment Using an Activated Sludge Process.
+United States: N.p., 21 Jan, 2022. Web. doi: 10.7481/1844539.
+"""
+import math
+import pyomo.environ as pyo
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.misc import StrEnum
+from ..util import (
+    make_capital_cost_var,
+    make_fixed_operating_cost_var,
+    register_costing_parameter_block,
+)
+
+
+class ClarifierType(StrEnum):
+    circular = "circular"
+    rectangular = "rectangular"
+    primary = "primary"
+
+
+def cost_clarifier(blk, clarifier_type=ClarifierType.circular, **kwargs):
+    """
+    Clarifier costing method
+
+    Args:
+        clarifier_type: ClarifierType Enum indicating clarifier type,
+            default = ClarifierType.circular
+    """
+    if clarifier_type == clarifier_type.circular:
+        cost_circular_clarifier(blk, **kwargs)
+    elif clarifier_type == clarifier_type.rectangular:
+        cost_rectangular_clarifier(blk, **kwargs)
+    elif clarifier_type == clarifier_type.primary:
+        cost_primary_clarifier(blk, **kwargs)
+    else:
+        raise ConfigurationError(
+            f"{blk.unit_model.name} received invalid argument for clarifier_type:"
+            f" {clarifier_type}. Argument must be a member of the ClarifierType Enum."
+        )
+
+
+def build_circular_clarifier_cost_param_block(blk):
+
+    blk.construction_a_parameter = pyo.Param(
+        initialize=-6e-4,
+        doc="A parameter for construction cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**4,
+    )
+
+    blk.construction_b_parameter = pyo.Param(
+        initialize=98.952,
+        doc="B parameter for construction cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**2,
+    )
+
+    blk.construction_c_parameter = pyo.Param(
+        initialize=191806,
+        doc="C parameter for construction cost",
+        units=pyo.units.USD_2011,
+    )
+
+    blk.O_and_M_a_parameter = pyo.Param(
+        initialize=8e-10,
+        doc="A parameter for operation and maintenance cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**6 / pyo.units.year,
+    )
+
+    blk.O_and_M_b_parameter = pyo.Param(
+        initialize=-5e-5,
+        doc="B parameter for operation and maintenance cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**4 / pyo.units.year,
+    )
+
+    blk.O_and_M_c_parameter = pyo.Param(
+        initialize=1.6945,
+        doc="C parameter for operation and maintenance cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**2 / pyo.units.year,
+    )
+
+    blk.O_and_M_d_parameter = pyo.Param(
```

Comment:
THese should be `Vars`, not `Params`. The underlying function (`register_costing_parameter_block`) will fix all `Vars`, if I recall correctly.